### PR TITLE
Update test-functional.yml - Fix vulnerability code injection

### PR DESCRIPTION
**Pull request by h2oa**
## Description

Please include a concise summary, in clear English, of the changes in this pull request. If it closes an issue, please mention it here.

Closes: #(issue)

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: .github/workflows/test-functional.yml [hunk 1]
```diff
@@ -34,7 +34,11 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
       - name: echo github context
         if: always()
-        run: echo "${{ github.event.workflow_run.head_repository.full_name }}-${{ github.event.workflow_run.head_branch }}-${{ github.workflow_ref }}"
+        env:
+          FULL_NAME: ${{ github.event.workflow_run.head_repository.full_name }}
+          HEAD_BRANCH: ${{ github.event.workflow_run.head_branch }}
+          WORKFLOW_REF: ${{ github.workflow_ref }}
+        run: echo "$FULL_NAME-$HEAD_BRANCH-$WORKFLOW_REF"
```
Comment (pngwn) [inline .github/workflows/test-functional.yml:42]: We can actually just delete the offending `echo`, it is unused and a silly leftover from testing.

**Review by pngwn — CHANGES_REQUESTED** [review]
Thanks for this PR! Just left a small comment.

**Comment by freddyaboulton** [comment]
Would love to merge this in @h2oa ! Can you make the small change @pngwn suggested ?

File: .github/workflows/test-functional.yml [hunk 2]
```diff
@@ -34,7 +34,10 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
       - name: echo github context
         if: always()
-        run: echo "${{ github.event.workflow_run.head_repository.full_name }}-${{ github.event.workflow_run.head_branch }}-${{ github.workflow_ref }}"
+        env:
```
Comment (freddyaboulton) [inline .github/workflows/test-functional.yml:37]: I think we can get rid of this entire step actually

**Review by pngwn — APPROVED** [review]
Thanks for this @h2oa!

**Pull request merged**